cpp/daal/src/algorithms/k_nearest_neighbors/bf_knn_impl.i

```diff
@@ -256,6 +262,7 @@ protected:
             FPType * distances = distancesBlock.get();
 
             const size_t size = blockSize * k * sizeof(FPType);
+            DAAL_OVERFLOW_CHECK_BY_MULTIPLICATION(size_t, blockSize * sizeof(FPType), k);
```

Comment:
I think you can do this check before the for.

Reply:
Fixed